### ISSUE TITLE
traffic_ctl: Timeout connection to jsonrpc socket

### ISSUE THREAD

**Issue by cmcfarlen** [apache/trafficserver#11468]
I had a situation where traffic_server was unhappy but some tools were trying to run traffic_ctl.  These all blocked on the connect to the jsonrpc socket and ended up with tons of traffic_ctl instances (and too many open fd errors).  Could/should we add a timeout to that connect?  The tool running traffic_ctl was also misbehaving, but this would be nice to have.
